samples/Microsoft.Bot.Samples.EchoBot-AspNet461/App_Start/WebApiConfig.cs

```diff
@@ -38,10 +38,11 @@ public static void Register(HttpConfiguration config)
             // services
             UnityConfig.Container.RegisterSingleton<BotFrameworkAdapter>(
                 new Unity.Injection.InjectionConstructor(
-                    new BotFrameworkAdapter(ConfigurationManager.AppSettings[@"MicrosoftAppId"], ConfigurationManager.AppSettings[@"MicrosoftAppPassword"])
-                        .Use(new BotStateManager(new MemoryStorage()))
+                    ConfigurationManager.AppSettings[@"MicrosoftAppId"],
+                    ConfigurationManager.AppSettings[@"MicrosoftAppPassword"])
                 )
-            );
+            .Resolve<BotFrameworkAdapter>()
+            .Use(new BotStateManager(new MemoryStorage()));

```

Comment:
Note: This is no longer correct. The BotStateManager is gone, in place of the UserStateManagerMiddleware and the ConversationStateManagerMiddleware.

Reply:
pullin' the rug out from under me! ;)